src/sentry/notifications/manager.py

```diff
@@ -0,0 +1,282 @@
+from django.db import transaction
+
+from sentry.db.models import BaseManager
+from sentry.models.integration import ExternalProviders
+from sentry.notifications.types import (
+    NotificationScopeType,
+    NotificationSettingOptionValues,
+    NotificationSettingTypes,
+)
+from sentry.models.useroption import UserOption
+from sentry.notifications.legacy_mappings import KEYS_TO_LEGACY_KEYS, KEY_VALUE_TO_LEGACY_VALUE
+
+
+def validate(type: NotificationSettingTypes, value: NotificationSettingOptionValues):
+    """
+    :return: boolean. True if the "value" is valid for the "type".
+    """
+    return _get_legacy_value(type, value) is not None
+
+
+def _get_scope(user_id, project=None, organization=None):
+    """
+    Figure out the scope from parameters and return it as a tuple,
+    TODO(mgaeta): Make sure user_id is in the project or organization.
+    :param user_id: The user's ID
+    :param project: (Optional) Project object
+    :param organization: (Optional) Organization object
+    :return: (int, int): (scope_type, scope_identifier)
+    """
+
+    if project:
+        return NotificationScopeType.PROJECT.value, project.id
+
+    if organization:
+        return NotificationScopeType.ORGANIZATION.value, organization.id
+
+    if user_id:
+        return NotificationScopeType.USER.value, user_id
+
+    raise Exception("scope must be either user, organization, or project")
+
+
+def _get_target(user=None, team=None):
+    """
+    Figure out the target from parameters and return it as a tuple.
+    :return: (int, int): (target_type, target_identifier)
+    """
+
+    if user:
+        return user.actor
+
+    if team:
+        return team.actor
+
+    raise Exception("target must be either a user or a team")
+
+
+def _get_legacy_key(type: NotificationSettingTypes):
+    """
+    Temporary mapping from new enum types to legacy strings.
+    :param type: NotificationSettingTypes enum
+    :return: String
+    """
+
+    return KEYS_TO_LEGACY_KEYS.get(type)
+
+
+def _get_legacy_value(type: NotificationSettingTypes, value: NotificationSettingOptionValues):
+    """
+    Temporary mapping from new enum types to legacy strings. Each type has a separate mapping.
+    :param type: NotificationSettingTypes enum
+    :param value: NotificationSettingOptionValues enum
+    :return: String
+    """
+
+    return str(KEY_VALUE_TO_LEGACY_VALUE.get(type, {}).get(value))
+
+
+class NotificationsManager(BaseManager):
+    """
+    TODO(mgaeta): Add a caching layer for notification settings
```

Comment:
is this for later or does it need to be addressed?

Reply:
We have another ticket for this in Jira.